hack/migrate-to-antrea.sh

```diff
@@ -0,0 +1,223 @@
+ANTREA_MANIFEST="./build/yamls/antrea.yml"
+
+while [[ $# -gt 0 ]]; do
+    key="$1"
+
+    case $key in
+    --from)
+        FROM="$2"
+        shift 2
+        ;;
+    --networkpolicy)
+        NETWORK_POLICY=true
+        shift
+        ;;
+    --dry-run)
+        DRY_RUN=true
+        shift
+        ;;
+    --back-up)
+        BACK_UP=true
+        shift
+        ;;
+    --output-dir)
+        OUTPUT_DIR="$2"
+        shift 2
+        ;;
+    --kubeconfig)
+        KUBECONFIG="$2"
+        shift 2
+        ;;
+    -h | --help)
+        print_usage
+        exit 0
+        ;;
+    *) # unknown option
+        echo "Unknown option $1"
+        exit 1
+        ;;
+    esac
+done
+
+if [ "$FROM" == "calico" ]; then
+    echo "$_calico_warning_message"
+    # input to continue
+    echo "========== Input 'yes' to continue =========="
+    read -r input
+    if [ "$input" != "yes" ]; then
+        echo "========== Aborting =========="
+        exit 1
+    fi
+fi
+echo "========== Building antctl =========="
+make antctl
+mv ./bin/antctl-${PLATFORM} ./bin/antctl
+
+echo "========== Checking the environment =========="
+if [ $DRY_RUN == true ]; then
+    echo "./bin/antctl migrate check-env --from $FROM"
+else
+    ./bin/antctl migrate check-env --from $FROM
+    if [ $? -ne 0 ]; then
+        echo "Failed to check the environment"
+        exit 1
+    fi
+fi
+
+if [ "$backup" == true ]; then
+    if [ "$OUTPUT_DIR" == "" ]; then
+        echo "Please specify the output directory for the backup files"
+        exit 1
+    fi
+    echo "========== Backing up the original CNI configuration files and network policies =========="
+    if [ $DRY_RUN == true ]; then
+        echo "kubectl get networkpolicies.crd.projectcalico.org --all-namespaces --kubeconfig $KUBECONFIG -o yaml >$OUTPUT_DIR/networkpolicies.yaml"
+        echo "kubectl get globalnetworkpolicies --all-namespaces --kubeconfig $KUBECONFIG -o yaml >$OUTPUT_DIR/globalnetworkpolicies.yaml"
+    else
+        kubectl get networkpolicies.crd.projectcalico.org --all-namespaces --kubeconfig "$KUBECONFIG" -o yaml >"$OUTPUT_DIR"/networkpolicies.yaml
+        kubectl get globalnetworkpolicies --all-namespaces --kubeconfig "$KUBECONFIG" -o yaml >"$OUTPUT_DIR"/globalnetworkpolicies.yaml
+    fi
+fi
+
+echo "========== Deploying Antrea ==========
+========== Warning: The Service might be unavailable after deploying Antrea to your cluster ==========
+========== Please make sure that you want to continue ==========
+========== Input 'yes' to continue =========="
+read -r input
+if [ "$input" != "yes" ]; then
+    echo "========== Aborting =========="
+    exit 1
+fi
+
+if [ $DRY_RUN == true ]; then
+    echo "kubectl apply -f $ANTREA_MANIFEST --kubeconfig $KUBECONFIG"
+else
+    trap "kubectl delete -f $ANTREA_MANIFEST" ERR
+    kubectl apply -f $ANTREA_MANIFEST --kubeconfig "$KUBECONFIG"
+fi
+echo "========== Waiting for Antrea to be ready =========="
```

Comment:
As for the outputs, I also feel too many equal symbols make it not easy to read. I will look for a better way for the log output.

Reply:
Reformat the output message, now only every step will contain equal symbols such as `======= Step Title ======`, and other messages just print directly. What's more, the `--dry-run` mode will not print detailed messages now, it will only print the commands of every step.  Just as follows:

```bash
========== Building antctl ==========
make antctl
========== Checking the environment ==========
./bin/antctl migrate check-env --from calico
========== Deploying Antrea ==========
Warning: The Service might be unavailable after deploying Antrea to your cluster
Please make sure that you want to continue
Input 'yes' to continue

yes
kubectl apply -f ./build/yamls/antrea.yml --kubeconfig /Users/hjiajing/.kube/config
kubectl rollout status ds/antrea-agent -n kube-system --timeout=300s --kubeconfig /Users/hjiajing/.kube/config
========== Restarting the sandboxes ==========
kubectl apply -f ./build/yamls/antrea-migrator.yml --kubeconfig /Users/hjiajing/.kube/config
kubectl rollout status ds/antrea-migrator -n kube-system --timeout=300s --kubeconfig /Users/hjiajing/.kube/config
========== The restart process is finished, begin to clean the old CNI ==========
kubectl delete -f https://docs.projectcalico.org/manifests/calico.yaml --kubeconfig /Users/hjiajing/.kube/config
========== Cleaning iptabes rules ==========
kubectl exec -n kube-system {ANTREA_AGENT} -- /bin/bash -c 'iptables-save | grep -v cali | iptables-restore' --kubeconfig /Users/hjiajing/.kube/config
========== Migration Completed ==========
```